From 2723331b45275fab9ba32dd869fbd7721bc2a887 Mon Sep 17 00:00:00 2001
From: Yong Qin <yong.qin@amlogic.com>
Date: Wed, 25 Dec 2019 17:31:03 +0800
Subject: [PATCH 6/7] vdin: transfer vsif info at event call back [2/2]

PD#SWPL-18998

Problem:
DV need vsif from hdmirx, vdin pass vsif data to DV
module.

Solution:
transfer vsif pointer and  size to dv

Verify:
tm2

Change-Id: I66821997292f79062bf3101f51207cc8418171a8
Signed-off-by: Yong Qin <yong.qin@amlogic.com>
---
 drivers/amlogic/media/vin/tvin/vdin/vdin_ctl.c |  4 +++-
 drivers/amlogic/media/vin/tvin/vdin/vdin_drv.h |  2 +-
 include/linux/amlogic/media/vfm/vframe.h       | 12 ++++++++++++
 3 files changed, 16 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/media/vin/tvin/vdin/vdin_ctl.c b/drivers/amlogic/media/vin/tvin/vdin/vdin_ctl.c
index 5cec6321a298..633ebfda7814 100644
--- a/drivers/amlogic/media/vin/tvin/vdin/vdin_ctl.c
+++ b/drivers/amlogic/media/vin/tvin/vdin/vdin_ctl.c
@@ -4605,7 +4605,6 @@ int vdin_event_cb(int type, void *data, void *op_arg)
 		req->low_latency = p->low_latency;
 		memcpy(&req->dv_vsif,
 			&p->dv_vsif, sizeof(struct tvin_dv_vsif_s));
-
 		if (req->bot_flag)
 			index = (req->vf->index >> 8) & 0xff;
 		if (index != 0xff
@@ -4844,6 +4843,9 @@ void vdin_set_drm_data(struct vdin_dev_s *devp,
 	devp->parm.info.signal_type = vf->signal_type;
 	/*hdr10+ check*/
 	vdin_hdr10plus_check(devp, vf);
+
+	vf->vsif.addr = &devp->prop.dv_vsif_raw;
+	vf->vsif.size = sizeof(struct tvin_dv_vsif_raw_s);
 }
 
 
diff --git a/drivers/amlogic/media/vin/tvin/vdin/vdin_drv.h b/drivers/amlogic/media/vin/tvin/vdin/vdin_drv.h
index f20a093c28c1..857145552c74 100644
--- a/drivers/amlogic/media/vin/tvin/vdin/vdin_drv.h
+++ b/drivers/amlogic/media/vin/tvin/vdin/vdin_drv.h
@@ -48,7 +48,7 @@
 /* Ref.2019/04/25: tl1 vdin0 afbce dynamically switch support,
  *                 vpp also should support this function
  */
-#define VDIN_VER "ver:2019-1118: 1080p 10bit afbc green screen"
+#define VDIN_VER "ver:2019-1225: transfer vsif to dv module"
 
 /*the counter of vdin*/
 #define VDIN_MAX_DEVS			2
diff --git a/include/linux/amlogic/media/vfm/vframe.h b/include/linux/amlogic/media/vfm/vframe.h
index 0fe6e63cb313..59ff4ec72e40 100644
--- a/include/linux/amlogic/media/vfm/vframe.h
+++ b/include/linux/amlogic/media/vfm/vframe.h
@@ -331,6 +331,16 @@ struct codec_mm_box_s {
 	int     bmmu_idx;
 };
 
+struct vsif_info {
+	void *addr;
+	unsigned int size;
+};
+
+struct emp_info {
+	void *addr;
+	unsigned int size;
+};
+
 struct vframe_s {
 	u32 index;
 	u32 index_disp;
@@ -454,6 +464,8 @@ struct vframe_s {
 	u32 di_gmv;
 
 	struct codec_mm_box_s mm_box;
+	struct vsif_info vsif;
+	struct emp_info emp;
 } /*vframe_t */;
 
 #if 0
-- 
2.24.1

